File: PE014/C++/Brute Force/PE014.cpp
#include<iostream>
#include<ctime>
using namespace std;


unsigned long even(unsigned long n);
unsigned long odd(unsigned long n);
unsigned long loop(unsigned long n);
 



int main()
{
    clock_t start,stop;
    unsigned long number =13;
    unsigned long counter = 0;
    unsigned long max=0;
    unsigned long max_number=0;
    
    start=clock();
    for(number=1;number<=1000000;number++)
    {
        counter=loop(number);
        if(counter>max)
        {
            max=counter;
            max_number=number;
        }
    }

    stop=clock();
    cout<<((stop-start)/(double)CLOCKS_PER_SEC)<<endl;
    cout<<"number:  "<<max_number<<"  chain:  "<<max<<endl;
}

unsigned long even(unsigned long n)
{
    n/=2;
    return n;   
}

unsigned long odd(unsigned long n)
{
    n*=3;
    n+=1;
    return n;   
}

unsigned long loop(unsigned long number)
{
    unsigned long counter = 1;

    do{
            if(number%2==0 && number!=1)
            {   
                counter+=1;
                number=even(number);
                //cout<<number<<endl;
            }
            if(number%2!=0 && number!=1)
            {
                counter+=1;
                number=odd(number);
                //cout<<number<<endl;   
            } 
        }while(number!=1);
        
        return counter;
}

File: PE014/Python/Brute Force/PE014.py
from datetime import datetime

#variables that i will use in this program
passed=0 #used in all of the functions
chain=0 #the max chain number
chain_number=0 #number that results in the max chain
current_chain=0 #the value of the current chain
number=0 #used as a loop variable

#functions for this program

def even(passed):
    passed/=2
    return passed

def odd(passed):
    passed*=3
    passed+=1
    return passed

def loop(passed):
    counter=1
    while passed!=1:
        if passed%2==0:
            counter+=1
            passed=even(passed)
        elif passed%2==1:
            counter+=1
            passed=odd(passed)
    return counter

startTime = datetime.now()

for number in xrange(1,1000000):
    current_chain=loop(number)
    if current_chain>chain:
        chain=current_chain
        chain_number=number

print chain, "  ", chain_number 
print(datetime.now()-startTime)
